fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -512,14 +515,33 @@ private void handleShowDb() throws AnalysisException {
                     PrivPredicate.SHOW)) {
                 continue;
             }
-
             dbNameSet.add(db);
         }
 
         for (String dbName : dbNameSet) {
             rows.add(Lists.newArrayList(dbName));
         }
 
+        if (showDbStmt.getDb() != null) {
+            rows.clear();
+            String db = showDbStmt.getDb();
+            MetadataMgr metadataMgr = GlobalStateMgr.getCurrentState().getMetadataMgr();
+            // get databases from external catalog
+            // eg: for hive, get db info from hms
+            // if fe can't connect to hms, would throw DdlException
+            try {
+                Optional<ConnectorMetadata> connectorMetadata = metadataMgr.getOptionalMetadata(db);
+                if (connectorMetadata != null) {
+                    List<String> externalDbNames = connectorMetadata.get().listDatabaseNames();
+                    for (String dbName : externalDbNames) {
+                        rows.add(Lists.newArrayList(dbName));
+                    }
+                }
+            } catch (NullPointerException e) {
```

Comment:
why catch NullPointerException? I think it is not common to catch NullPointer.

Reply:
> why catch NullPointerException? I think it is not common to catch NullPointer.

If execute dql `show databases from catalog_a`,while catalog_a doesn't exist, it would throw NullPointer here.

Reply:
why throw nullptr exception. I think a normal exception should be thrown. Such as AnalysisException.

Reply:
> 

refactored, now throws  AnalysisException, DdlException